core/src/main/java/hudson/Launcher.java

```diff
@@ -450,8 +450,14 @@ public ProcStarter writeStdin() {
 
         /**
          * Starts the new process as configured.
+         *
+         * @throws RuntimeException can be thrown if {@link #cmds()} is empty
          */
         public Proc start() throws IOException {
+            if(cmds().isEmpty()){
+                LOGGER.log(Level.INFO, "Trying to start a process without setting a command. This might cause "
+                                       + "an unexpected exception");
```

Comment:
Seems to be we want the stack trace here, otherwise how does this even help?

Reply:
The exception is not being caught.
Logging or not the exception is up to the client. If they want to handle it, logging it will only add noise to the logs. If they don't handle it, it's a RuntimeException, at some point, it will appear at the logs.

I'm detailing the exception a little bit thought